denonavr/denonavr.py

```diff
@@ -819,23 +816,45 @@ def _update_input_func_list(self):
     def _get_receiver_name(self):
         """Get name of receiver from web interface if not set."""
         # If name is not set yet, get it from Main Zone URL
-        if self._name is None and self._urls.mainzone is not None:
-            name_tag = {"FriendlyName": None}
-            try:
-                root = self.get_status_xml(self._urls.mainzone)
-            except requests.exceptions.ConnectTimeout:
-                pass
-            except (ValueError, requests.exceptions.RequestException):
-                _LOGGER.warning("Receiver name could not be determined. "
-                                "Using standard name: Denon AVR.")
-                self._name = "Denon AVR"
-            else:
-                # Get the tags from this XML
-                name_tag = self._get_status_from_xml_tags(root, name_tag)
-                if name_tag:
+        if self._name is None:
+            if self._receiver_type == AVR_X_2016.type:
```

Comment:
appcommand and status xml endpoints return exactly the same attribute. Thus there is no need to build two different cases.

Reply:
@scarface-4711 the point is that the appcommand was not implemented and AVR_X_2016.type does not have a status xml endpoint anymore as far as I know.
Therefore this would add the ability to get the friendlyName for avr-x-2016 receivers, which is currently not possible as far as I can tell